### DISCUSSION TITLE
socketio.wsgiApp staticfiles cors issues.

### DISCUSSION THREAD

**Discussion by TheDaneSimon** [miguelgrinberg/python-socketio#769] [Q&A] [answered]
i have a problem with cors and serving static files for wsgi, it seems to ignore the `cors_allowed_origins="*"`
the static files are served as follwing:

`sio = socketio.Server(async_mode='eventlet', cors_allowed_origins="*")`
`wsgiapp = socketio.WSGIApp(sio,static_files={"/files":"/home/vaerks/CNCProjects", 'content-type': 'application/blob'})`

My implementation of the served files are in react, and download the file as seen in the screen shot.

![Capture](https://user-images.githubusercontent.com/22538021/129729966-86207429-400f-45f9-abf7-193f9224f186.PNG)

but i get the follwing cors error when using the served file from the wsgiApp, as visable from the error it is not on the same computer, but over lan.

![cors error](https://user-images.githubusercontent.com/22538021/129730086-19ec3e4f-0b55-4c14-af7a-1e0296242241.PNG)

If i use a database i have with same cors allow *, then i get the file as expected in react.

the file is served, since i can access by putting in the file url in a browser and downloading the file, on the same computer that is hosting the react project.

i am running deprecated python-socketio (4.6.1) becauce i cant upgrade to python 3,5

eventlet (0.31.1)
python-engineio (3.14.2)
python-socketio (4.6.1)

**Comment by miguelgrinberg** [comment, marked as answer]
The static file serving app does not have CORS configuration, so it must be used within a same origin context, or else the CORS configuration must be inserted by a reverse proxy (nginx).

The purpose of serving static files is to aid in the development of your app by serving your client files. If you have needs that are larger than this, then I suggest you add your own file serving solution based on Flask or similar framework.
